WebApp/autoreduce_webapp/selenium_tests/tests/test_failed_jobs_page.py

```diff
@@ -0,0 +1,61 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+"""
+Selenium tests for failed jobs page
+"""
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium_tests.pages.failed_jobs_page import FailedJobsPage
+from selenium_tests.tests.base_tests import FooterTestMixin, BaseTestCase, NavbarTestMixin
+
+from WebApp.autoreduce_webapp.selenium_tests.utils import setup_external_services
+from model.database import access
+
+
+class TestFailedJobsPage(NavbarTestMixin, BaseTestCase, FooterTestMixin):
+    """
+    Test cases for Failed Jobs Page
+    """
+    fixtures = BaseTestCase.fixtures + ["failed_jobs_fixture"]
+
+    def setUp(self) -> None:
+        """
+        Sets up Failed Jobs Page model
+        """
+        super().setUp()
+        self.page = FailedJobsPage(self.driver)
+        self.page.launch()
+
+    def test_failed_runs_visible(self):
+        """
+        Test: Correct runs shown on table
+        """
+        self.assertCountEqual(["99999"], self.page.get_failed_run_numbers())
+
+    def test_hide_run_hides_run(self):
+        """
+        Test hide option hides run from page
+        """
+        self.page.toggle_run("99999").hide_runs()
+        self.assertEqual([], self.page.get_failed_run_numbers())
+
+    def test_rerun_failed_run(self):
+        """
+        Test rerunning failed run creates rerun
+        """
+
+        archive, db_client, queue_client, listener = setup_external_services("TestInstrument", 21, 21)
+        try:
+            listener._processing = True  # pylint:disable=protected-access
+            self.page.toggle_run("99999").retry_runs()
+            WebDriverWait(self.driver, 30).until(lambda _: not listener.is_processing_message())
+            runs = access.get_reduction_run("TestInstrument", "99999")
```

Comment:
Could this use [submit_and_wait_for_result](https://github.com/ISISScientificComputing/autoreduce/blob/625245cc80fe55005a5b977cb7d0a0fed05142a5/WebApp/autoreduce_webapp/selenium_tests/utils.py)?

Reply:
It can't as the url doesnt change and it redirects back to the original list and not a run confirmation.

Reply:
I think you could move the `expected_url` to be a function argument with `None` default, and `if expected_url is None` then it gets the reverse of `"run_confirmation"`. Would that work?